astropy/visualization/tests/test_norm.py

```diff
@@ -170,3 +171,27 @@ def test_invalid_stretch(self):
         """Test invalid stretch keyword."""
         with pytest.raises(ValueError):
             simple_norm(DATA2, stretch='invalid')
+
+@pytest.mark.skipif('not HAS_MATPLOTLIB')
+def testy_imshow_norm():
+    image = np.random.randn(10, 10)
+
+    ax = plt.subplot()
+    imshow_norm(image, ax=ax)
+
+    with pytest.raises(ValueError):
+        # X and data are the same, can't give both
+        imshow_norm(image, X=image, ax=ax)
+
+    with pytest.raises(ValueError):
+        # illegal to manually pass in normalization since that defeats the point
+        imshow_norm(image, ax=ax, norm=ImageNormalize())
+
+    # vmin/vmax "shadow" the MPL versions, so we allow direct-setting
+    imshow_norm(image, ax=ax, vmin=0, vmax=1)
+    imshow_norm(image, ax=ax, vmin_mpl=0, vmax_mpl=1)
```

Comment:
I'm not a fan of this way of passing arguments directly to Matplotlib. Just to understand, what's the motivation for this? If you want to pass things to Matplotlib bypassing ImageNormalize, maybe you should just not use imshow_norm?

Reply:
Ditto.

Reply:
Ah, the idea is that I can't guarantee that `imshow` won't in the future get, say a `stretch` keyword, or `ImageNormalize` might get some other keywords that conflicts with matplotlib (the latter we control of course, but someone might forget...).  So the machinery is there to account for such conflicts.  

But currently, the only one I saw is `vmin`/`vmax`.  I agree it's not reasonable for a user to ever do this, but it's a way to test the `_mpl` machinery so that the function is future-proofed.

I could catch the `vmin`/`vmax` cases and raise an exception if you think that's better, @astrofrog or @larrybradley  (like with the `norm` keyword case - https://github.com/astropy/astropy/pull/7785/files#diff-3163f2e650a29867e5d840ff276cc2a0R261), but then it's not possible to test the `_mpl` shadowing trick right now...

Reply:
Since this is not meant to be used widely, what if you just had a ``imshow_kwargs`` argument for arguments that should be passed directly to ``imshow``? Then other arguments would be interpreted first by ``imshow_norm``, and passed to ``imshow`` only if they haven't already been used.

Reply:
OK, I see your point here, @astrofrog - too clever by half here... :wink:  I'm pretty sure where you said "interpreted first by `imshow_norm`" you meant "interpreted first by `ImageNormalize`", so I'll go that route